src/synopsys-action/tools-parameter.ts

```diff
@@ -218,40 +218,51 @@ export class SynopsysToolsParameter {
     return command
   }
 
-  private setGithubData(): GithubData | undefined {
+  private getGithubRepoInfo(): GithubData | undefined {
     const githubToken = inputs.GITHUB_TOKEN
     const githubRepo = process.env[FIXPR_ENVIRONMENT_VARIABLES.GITHUB_REPOSITORY]
     const githubRepoName = githubRepo !== undefined ? githubRepo.substring(githubRepo.indexOf('/') + 1, githubRepo.length).trim() : ''
     const githubBranchName = process.env[FIXPR_ENVIRONMENT_VARIABLES.GITHUB_HEAD_REF]
     const githubRef = process.env[FIXPR_ENVIRONMENT_VARIABLES.GITHUB_REF]
     // pr number will be part of "refs/pull/<pr_number>/merge"
-    const githubPrNumber = githubRef !== undefined ? githubRef.split('/')[2].trim() : null
+    // if there is manual run without raising pr then GITHUB_REF will return refs/heads/branch_name
+    const githubPrNumber = githubRef !== undefined ? githubRef.split('/')[2].trim() : ''
     const githubRepoOwner = process.env[FIXPR_ENVIRONMENT_VARIABLES.GITHUB_REPOSITORY_OWNER]
 
     if (githubToken == null) {
       throw new Error('Missing required github token for fix pull request/automation comment')
     }
 
+    if ((parseToBoolean(inputs.BLACKDUCK_AUTOMATION_PRCOMMENT) || parseToBoolean(inputs.COVERITY_AUTOMATION_PRCOMMENT)) && isNaN(Number(githubPrNumber))) {
+      throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
```

Comment:
Check with @Madhu if this error msg is fine " Pull request missing for Coverity or Blackduck automation PR comment. " We can modify accordingly.

Reply:
going with "Coverity/Blackduck automation PR comment can only be triggered on a pull request."